### PR TITLE
[BugFix] Fixed address Validation

### PR DESCRIPTION
**Pull request by neelSharma12**
This PR contains fix for billing address issue where the billing Address is coming as nil if the country does not have a state/province.

### CODE REVIEW

File: Adyen/UI/Form/Items/Address/AddressViewModel.swift [hunk 1]
```diff
@@ -17,7 +17,13 @@ internal enum AddressField: String, CaseIterable {
     case country
 }
 
-internal enum FormScheme {
+public enum FormScheme {
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/AddressViewModel.swift:22]: maybe add the `///:nodoc:`?

File: Adyen/UI/Form/Items/Address/AddressViewModel.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import Foundation
 import UIKit
 
-internal enum AddressField: String, CaseIterable {
+public enum AddressField: String, CaseIterable {
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/AddressViewModel.swift:11]: `///:nodoc:`? :-)

File: Adyen/UI/Form/Items/Address/AddressViewModel.swift [hunk 3]
```diff
@@ -27,12 +33,12 @@ internal struct AddressViewModelBuilderContext {
     internal var isOptional: Bool
 }
 
-internal struct AddressViewModel {
+public struct AddressViewModel {
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/AddressViewModel.swift:42]: `///:nodoc:`? :-)

File: Adyen/UI/Form/Items/Address/AddressViewModel.swift [hunk 4]
```diff
@@ -17,7 +17,13 @@ internal enum AddressField: String, CaseIterable {
     case country
 }
 
-internal enum FormScheme {
+public enum FormScheme {
+    public var  children: [AddressField] {
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/AddressViewModel.swift:24]: `///:nodoc:`? :-)

File: Adyen/UI/Form/Items/Address/AddressViewModel.swift [hunk 5]
```diff
@@ -27,12 +33,12 @@ internal struct AddressViewModelBuilderContext {
...
 
-internal struct AddressViewModel {
+public struct AddressViewModel {
 
     internal var labels: [AddressField: LocalizationKey]
     internal var placeholder: [AddressField: LocalizationKey]
-    internal var optionalFields: [AddressField]
-    internal var schema: [FormScheme]
+    public var optionalFields: [AddressField]
+    public var schema: [FormScheme]
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/AddressViewModel.swift]: `///:nodoc:` for both properties? :-)

also would be great to have them `private(set)` or at least `internal(set)`

File: Adyen/UI/Form/Items/Address/FormFullAddressItem.swift [hunk 6]
```diff
@@ -31,6 +31,8 @@ public final class FormAddressItem: FormValueItem<PostalAddress, AddressStyle>,
     override public var subitems: [FormItem] { items }
     
     internal let supportedCountryCodes: [String]?
+
+    public private(set) var addressViewModel: AddressViewModel
```
Comment (mohammedDehairy) [inline Adyen/UI/Form/Items/Address/FormFullAddressItem.swift:36]: `///:nodoc:`? :-)

**Review by mohammedDehairy — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Pull request merged**